registry/api/protocol/amalgam8/instance_handler.go

```diff
@@ -72,7 +72,22 @@ func (routes *Routes) registerInstance(w rest.ResponseWriter, r *rest.Request) {
 			"error":     err,
 		}).Warnf("Failed to register instance %+v", req)
 
-		i18n.Error(r, w, statusCodeFromError(err), i18n.ErrorInstanceRegistrationFailed)
+		if regerr, ok := err.(*store.Error); ok {
+			switch regerr.Code {
+			case store.ErrorNoInstanceServiceName:
+				i18n.Error(r, w, statusCodeFromError(err), i18n.ErrorNoServiceName)
+			case store.ErrorInstanceServiceNameTooLong:
+				i18n.Error(r, w, statusCodeFromError(err), i18n.ErrorServiceNameTooLong, serviceNameMaxLength)
+			case store.ErrorInstanceEndpointValueTooLong:
+				i18n.Error(r, w, statusCodeFromError(err), i18n.ErrorEndpointValueTooLong, valueMaxLength)
+			case store.ErrorInstanceStatusLengthTooLong:
+				i18n.Error(r, w, statusCodeFromError(err), i18n.ErrorStatusLengthTooLong, statusMaxLength)
+			case store.ErrorInstanceMetaDataTooLong:
+				i18n.Error(r, w, statusCodeFromError(err), i18n.ErrorMetaDataTooLong, metadataMaxLength)
+			default:
+				i18n.Error(r, w, statusCodeFromError(err), i18n.ErrorInstanceRegistrationFailed)
+			}
+		}
```

Comment:
It would be safer to add an `else` block here, for the case that `regerr` is not of type `*store.Error`. In that case, we would want to return the generic `i18n.ErrorInstanceRegistrationFailed` error message as well.